### PR TITLE
[ONE] Implement Noxious assault

### PR DESCRIPTION
**Pull request by AhmadYProjects**

### CODE REVIEW

**Comment by AhmadYProjects** [comment]
[[Noxious Assault]]

**Comment by AhmadYProjects** [comment]
see last commit comment

File: Mage.Sets/src/mage/cards/n/NoxiousAssault.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        Permanent permanent = game.getPermanent(event.getTargetId());
+        Player player = game.getPlayer(permanent.getControllerId());
+        if (permanent == null){
+            return false;
+        }
+        player.addCounters(CounterType.POISON.createInstance(1),permanent.getControllerId(),this,game);
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/n/NoxiousAssault.java:66]: why are you adding a counter here? this is where the trigger is checked, not where it resolves

File: Mage.Sets/src/mage/cards/n/NoxiousAssault.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    @Override
+    public boolean checkEventType(GameEvent event, Game game) {
+        return event.getType() == GameEvent.EventType.CREATURE_BLOCKS;
+    }
+
+    @Override
+    public boolean checkTrigger(GameEvent event, Game game) {
+        Permanent permanent = game.getPermanent(event.getTargetId());
+        Player player = game.getPlayer(permanent.getControllerId());
+        if (permanent == null){
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/n/NoxiousAssault.java:63]: if you're checking for null then do it before you call `permanent.getControllerId()`

File: Mage.Sets/src/mage/cards/n/NoxiousAssault.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    @Override
+    public NoxiousAssaultDelayedTriggerAbility copy(){
+        return new NoxiousAssaultDelayedTriggerAbility();
+    }
+
+    @Override
+    public String getRule(){
+        return "Whenever a creature blocks this turn, it's controller gets a poison counter.";
```
Comment (theelk801) [inline Mage.Sets/src/mage/cards/n/NoxiousAssault.java:77]: "it's"->"its"

**Review by theelk801 — CHANGES_REQUESTED** [review]

File: Mage.Sets/src/mage/cards/n/NoxiousAssault.java (hunk 3 above)
Comment (AhmadYProjects) [reply Mage.Sets/src/mage/cards/n/NoxiousAssault.java]: :O 

fixed

File: Mage.Sets/src/mage/cards/n/NoxiousAssault.java (hunk 2 above)
Comment (AhmadYProjects) [reply Mage.Sets/src/mage/cards/n/NoxiousAssault.java:62]: removed line 62 completely in new commits

File: Mage.Sets/src/mage/cards/n/NoxiousAssault.java (hunk 1 above)
Comment (AhmadYProjects) [reply Mage.Sets/src/mage/cards/n/NoxiousAssault.java]: Initially the effect wasn't popping off but I had forgotten to add a "this" when calling(i think this is the correct terminology right?) the ability

**Review by theelk801 — APPROVED** [review]

**Pull request merged**